### ISSUE TITLE
Development Setup

### ISSUE THREAD

**Issue by wedgex** [vaxinate/dndchar#7]
What all do I need, had to add coffee-script to the package.json locally. But I get the following when I run grunt.

``` Shell
hunter ~/dev/dndchar (master) ⚡  grunt

module.js:340
    throw err;
          ^
Error: Cannot find module 'grunt-legacy-util'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/Users/hunter/dev/dndchar/node_modules/grunt/lib/grunt.js:26:12)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
```

**Comment by vaxinate** [comment]
make sure your node is recent. I had trouble when i was trying to use an 0.8.x version. updated to 0.10.x and all was well. You shouldn't need coffeescript at all. sadly the yeoman generator I used doesn't support coffeescript and i didn't feel like figuring out how to add it in. the yeoman generator was kind of a mistake, but i'd already made that bed, so whatever.

**Comment by vaxinate** [comment]
make sure you do npm and bower install too.

**Comment by wedgex** [comment]
What version of node are you using?

``` Shell
npm WARN prefer global bower@1.3.3 should be installed with -g
npm ERR! peerinvalid The package bower does not satisfy its siblings' peerDependencies requirements!
npm ERR! peerinvalid Peer grunt-bower-requirejs@0.4.4 wants bower@~0.x

npm ERR! System Darwin 13.2.0
npm ERR! command "/Users/hunter/.nvm/v0.10.13/bin/node" "/Users/hunter/.nvm/v0.10.13/bin/npm" "install" "bower"
npm ERR! cwd /Users/hunter/dev/dndchar
npm ERR! node -v v0.10.13
npm ERR! npm -v 1.3.2
npm ERR! code EPEERINVALID
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!     /Users/hunter/dev/dndchar/npm-debug.log
npm ERR! not ok code 0
```

**Comment by vaxinate** [comment]
0.10.26 I think. 0.8.26 gave me similar problems to that.

**Comment by wedgex** [comment]
`npm ERR! peerinvalid Peer grunt-bower-requirejs@0.4.4 wants bower@~0.x`

Something seems weird right there. grunt bower wants an old version of
bower. I have 1.3.3 installed. It wants an 0.x version.

**Comment by vaxinate** [comment]
wait, are you running `grunt bower` just run `bower install` I'm not really a big fan of Grunt so far :-/

**Comment by vaxinate** [comment]
so far `grunt server` has been the only useful part of it.

**Comment by wedgex** [comment]
No I ran bower install. That acutally worked fine. npm install. and grunt server are being shitty.

**Comment by vaxinate** [comment]
can't help but feel like this is a problem with your node install. you using nodenv?

**Comment by wedgex** [comment]
nvm. I dug through some project issues and got a bit further, now the server is just failing to start due to some warnings.

**Comment by wedgex** [comment]
The phantomjs repo is down... https://bitbucket.org/ariya/phantomjs lol.

**Comment by vaxinate** [comment]
ugh. using this yeoman generator was a mistake. some of these problems are stemming from it offering up sass, which apparently is a real pain in the dick with node. luckily, the app only relies on node for the build system which i've luckily not needed at all.

**Comment by wedgex** [comment]
> node

FOUND YOUR PROBLEM :trollface:

**Comment by wedgex** [comment]
But for real tho, sass is what its complaining about

```
    Warning: /Users/hunter/.rbenv/versions/2.0.0-p195/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:110:in `require': cannot load such file -- sass/script/node (LoadError)
```

**Comment by wedgex** [comment]
Actually wth is going on, sup with the ruby error.

**Comment by vaxinate** [comment]
make sure you have the compass gem installed. that's what it wants to use to compile sass

**Comment by vaxinate** [comment]
So bower dependencies are updated. You added a Gemfile. Everything else was just your node being goofy?

**Comment by vaxinate** [comment]
reopen it if you need to.

**Comment by wedgex** [comment]
The your lodash config was off. I fixed it in #8. Seems like that's it. Now I need to relearn backbone.
